apps/researcher/src/app/[locale]/tabs.tsx

```diff
@@ -1,24 +1,37 @@
-'use client';
-
 import classNames from 'classnames';
 import {ObjectIcon, PersonIcon} from '@/components/icons';
-import {usePathname} from 'next-intl/client';
 import {useTranslations} from 'next-intl';
+import {headers} from 'next/headers';
+import {locales} from '@/middleware';
 
 export default function Tabs() {
   const t = useTranslations('Tabs');
-  const pathname = usePathname();
+
+  const activePath = headers().get('x-pathname') || '/';
+
   const tabs = [
-    {name: t('objects'), href: '/', icon: ObjectIcon},
-    {name: t('persons'), href: '/persons', icon: PersonIcon},
+    {
+      name: t('objects'),
+      isCurrentRegex: `^/(${locales.join('|')})?$`,
```

Comment:
Could, at some point, the locales contain characters that have meaning in regular expressions? If so, this could will break. Escape the locales?

Reply:
Good question. The locales are based on official language codes. Non contain characters that have meaning in regular expressions. So I will leave it like this.

https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes

Reply:
Perhaps you could add a comment to the code? Then it's clear why the locale values aren't escaped?